alternator/auth.cc

```diff
@@ -51,7 +53,14 @@ future<std::string> get_key_from_roles(service::storage_proxy& proxy, auth::serv
     if (result_set->empty()) {
         co_await coroutine::return_exception(api_error::unrecognized_client(format("User not found: {}", username)));
     }
-    const managed_bytes_opt& salted_hash = result_set->rows().front().front(); // We only asked for 1 row and 1 column
+    const auto& result = result_set->rows().front();
+    bool can_login = result[1] && value_cast<bool>(boolean_type->deserialize(*result[1]));
```

Comment:
wrt result[1], is it ok to assume result rows will match query specification in terms of columns returned?  guessing yes

Reply:
Good question, and I was puzzled by this myself and couldn't really prove it to myself. I am *hoping* we can indeed assume that if we asked for two columns, we get both of them back in the same order. The reason why I felt sort-of-comfortable with it was that the pre-existing code already assumed that if we asked for one column, we get it back (the code called front() without checking if the vector was empty before doing it). This value of each column can be null, but I **think** it exists in the vector.

Reply:
for order it should be ok, looks like we're already coupling it in executor::describe_single_item  - column_it is increamented with each result_row element